server/tracedb/datasource/http.go

```diff
@@ -27,6 +27,10 @@ type HttpClient struct {
 
 func NewHttpClient(name string, config *datastore.HttpClientConfig, callback HttpCallback) DataSource {
 	endpoint, _ := urlx.Parse(config.Url)
+	if endpoint.Port() == "443" {
+		endpoint.Scheme = "https"
+	}
```

Comment:
With a url like this:

`tempo-us-central1.grafana.net:443`

The parsing method returns

`http://tempo-us-central1.grafana.net:443`

Which breaks the `getTraceById` logic.